MSOE.MediaComplete.Lib/Library/DataSource/FileSystem.cs

```diff
@@ -567,17 +577,23 @@ private static bool IsFileLocked(FileInfo file)
         /// <param name="song"></param>
         private void UpdateFile(LocalSong song)
         {
-            var tagFile = TaglibFile.Create(song.Path);
-
-            var tag = tagFile.Tag;
-            _cachedSongs[song.Id].Title = tag.Title;
-            _cachedSongs[song.Id].Artists = tag.AlbumArtists;
-            _cachedSongs[song.Id].Album = tag.Album;
-            _cachedSongs[song.Id].Genres = tag.Genres;
-            _cachedSongs[song.Id].Year = tag.Year;
-            _cachedSongs[song.Id].TrackNumber = tag.Track;
-            _cachedSongs[song.Id].SupportingArtists = tag.Performers;
-            // Duration is assumed to be fixed
+            //Catch blocks do nothing. Sometimes this function is called on a file that has already been moved
+            //This happens if create & delete events happen AND a changed happens. Therefore, this does not have to execute
+            try
+            {
+                var tagFile = TaglibFile.Create(song.Path);
+                var tag = tagFile.Tag;
+                _cachedSongs[song.Id].Title = tag.Title;
+                _cachedSongs[song.Id].Artists = tag.AlbumArtists;
+                _cachedSongs[song.Id].Album = tag.Album;
+                _cachedSongs[song.Id].Genres = tag.Genres;
+                _cachedSongs[song.Id].Year = tag.Year;
+                _cachedSongs[song.Id].TrackNumber = tag.Track;
+                _cachedSongs[song.Id].SupportingArtists = tag.Performers;
+                // Duration is assumed to be fixed
+            }
+            catch (FileNotFoundException) {}
```

Comment:
Looks like my first comment went away after these lines changed- I think there should be a DEBUG statement in the catch blocks.